python/tvm/auto_scheduler/search_policy.py

```diff
@@ -61,6 +62,71 @@ def __init__(self, filename):
         self.__init_handle_by_constructor__(_ffi_api.PreloadMeasuredStates, filename)
 
 
+@tvm._ffi.register_object("auto_scheduler.PreloadCustomSketchRule")
+class PreloadCustomSketchRule(SearchCallback):
+    """
+    A SearchCallback for SketchSearchPolicy that allowing users to add
+    custom sketch rule.
+
+    Notes
+    -----
+    This is an advanced feature. Make sure you're clear how it
+    works and this should only be used in SketchSearchPolicy.
+
+    Parameters
+    ----------
+    meet_condition_func: Function
+        A function with `(policy, state, stage_id) -> int`
+    apply_func: Function
+        A function with `(policy, state, stage_id) -> [[State, int], ...]`
+    """
+
+    CONDITION_NUM = {"pass": 0, "apply": 1, "apply_and_skip_rest": 2}
+
+    def __init__(self, meet_condition_func, apply_func, rule_name="CustomSketchRule"):
+        self.__init_handle_by_constructor__(
+            _ffi_api.PreloadCustomSketchRule, meet_condition_func, apply_func, rule_name
+        )
+
+
+CUSTOM_SKETCH_REGISTRY = {}
+
+
+def register_custom_sketch_func(compute_name, func=None):
+    """Helper decorator to register custom sketch functions easily."""
+    global CUSTOM_SKETCH_REGISTRY
+
+    if callable(compute_name):
+        func = compute_name
+        compute_name = func.__name__
+
+    if not isinstance(compute_name, str):
+        raise ValueError("expect string function name")
+
+    def register(myf):
+        if compute_name in CUSTOM_SKETCH_REGISTRY:
+            raise RuntimeError(
+                "Custom Sketch for %s has been registered for compute already" % compute_name
+            )
+
+        def meet_condition_func(policy, state, stage_id):
+            state = State(state, policy.search_task.compute_dag)
+            if state.stages[stage_id].op.name == compute_name:
+                return PreloadCustomSketchRule.CONDITION_NUM["apply_and_skip_rest"]
+            return PreloadCustomSketchRule.CONDITION_NUM["pass"]
```

Comment:
Do not use string. Use  `PreloadCustomSketchRule.CONDITION_NUM.APPLY_AND_SKIP_REST` instead.
See our existing convention.
https://github.com/apache/tvm/blob/384714b58ed374cb1e385142b5dc4128041c945c/python/tvm/auto_scheduler/measure.py#L581-L587